packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -1,33 +1,249 @@
 /*
- * Copyright (c) 2017, salesforce.com, inc.
- * All rights reserved.
- * Licensed under the BSD 3-Clause license.
- * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
- */
+* Copyright (c) 2017, salesforce.com, inc.
+* All rights reserved.
+* Licensed under the BSD 3-Clause license.
+* For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+*/
 import * as fs from 'fs';
+import * as mkdirp from 'mkdirp';
 import * as path from 'path';
-import { SObjectCategory, SObjectDescribe } from '../describe';
+import {
+  ChildRelationship,
+  Field,
+  SObject,
+  SObjectCategory,
+  SObjectDescribe
+} from '../describe';
 
 export class FauxClassGenerator {
-  public async generate(projectPath: string, type: SObjectCategory) {
+  private SFDX_DIR = '.sfdx';
+  private TOOLS_DIR = 'tools';
+  private SOBJECTS_DIR = 'sobjects';
+
+  private createIfNeededOutputFolder(folderPath: string): boolean {
+    if (!fs.existsSync(folderPath)) {
+      mkdirp.sync(folderPath);
+      return fs.existsSync(folderPath);
+    }
+    return true;
+  }
+
+  public async generate(
+    projectPath: string,
+    type: SObjectCategory
+  ): Promise<string> {
     const describe = new SObjectDescribe();
     const sobjects = await describe.describeGlobal(projectPath, type);
-    console.log(sobjects.length);
-    for (let i = 0; i < sobjects.length; i++) {
-      const sobject = await describe.describeSObject(projectPath, sobjects[i]);
-      console.log(sobject.name);
-      this.generateFauxClass(projectPath, sobject);
+    const standardSObjects: SObject[] = [];
+    const customSObjects: SObject[] = [];
+    let fetchedSObjects: SObject[] = [];
+    let j = 0;
+    while (j < sobjects.length) {
+      try {
+        fetchedSObjects = fetchedSObjects.concat(
+          await describe.describeSObjectBatch(projectPath, sobjects, j - 1)
```

Comment:
Why don't you just have this start at j = 0 rather than -1? It's weird to set lastProcessed to -1 then increment it during the batch call.

Reply:
Well, it is because the arrays are 0 based, so lastProcessed needs to be lower than 0..  But, I agree it reads funny, so I may change it to "nextToProcess" and 0 base.    Old habit from days where you could overflow the integer range easily.